### DISCUSSION TITLE
onerror event is not fired when job fails

### DISCUSSION THREAD

**Discussion by igorg-il** [jobrunr/jobrunr#743] [Q&A]
We are using jobrunr 6.1.2

Our Front End listens to EventSource **onerror** event of url: <ourServerUrl>/sse/jobs/<jobId> to monitor status of job.

If job on server fails immediately by throwing exception **onerror** event doesn't fire but if before failing we write ANY message to jobrunr log (throu JobContext) the event does fires.
This behavior is consistent.

As a result in first case we don't know when job finishes.
Any idea how to handle this case to know when job finishes with failure? (e.g. listen to some other event raher than onerror)?
We can't forse server to always write log in all jobs.

**Comment by rdehuyss** [comment]
I don't understand what you want to achieve, to be honest. **onerror** event is not used by JobRunr.

**Reply by igorg-il** [reply]
So how Front End can know when job is finished?
thank you

**Reply by rdehuyss** [reply]
By following the international agreed upon standards 😄 .
